lib/rules/prefer-arrow-callback.js

```diff
@@ -115,6 +115,17 @@ function getCallbackInfo(node) {
     throw new Error("unreachable");
 }
 
+/**
+* Checks whether a simple list of parameters contains any duplicates. This does not handle complex
+parameter lists (e.g. with destructuring), since complex parameter lists are a SyntaxError with duplicate
+parameter names anyway. Instead, it always returns `false` for complex parameter lists.
+* @param {ASTNode[]} paramsList The list of parameters for a function
+* @returns {boolean} `true` if the list of parameters contains any duplicates
+*/
+function hasDuplicateParams(paramsList) {
```

Comment:
This should probably be replaced with a function like the one described [here](/eslint/eslint/pull/7004#discussion_r76553312).

The current implementation will work in all cases where the function body has valid syntax, but it will fail in certain cases with complex parameter lists due to the parser bug described [here](/eslint/eslint/pull/7004#discussion_r76553858).

```js
foo(function(a, a = 5) { }) // the parser incorrectly considers this to be valid.

foo((a, a = 5) => { }) // the parser correctly considers this to be invalid.
```